src/ConcordiumNetSdk/Client.cs

```diff
@@ -0,0 +1,818 @@
+﻿using Concordium.V2;
+using Grpc.Core;
+using Grpc.Net.Client;
+
+namespace ConcordiumNetSdk;
+
+/// <summary>
+/// A client for interacting with the GRPC V2 API exposed by Concordium nodes.
+/// </summary>
+public class Client : IDisposable
+{
+    /// <summary>
+    /// The "internal" client instance generated from the V2 API protocol buffer definition.
+    /// </summary>
+    private readonly Queries.QueriesClient _client;
+
+    /// <summary>
+    /// The channel on which the client communicates with the Concodium node.
+    /// </summary>
+    private readonly GrpcChannel _grpcChannel;
+
+    /// <summary>
+    /// The configuration to use with the client.
+    /// </summary>
+    private readonly ClientConfiguration _config;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="Client"/> class.
+    /// Uses the <see cref="ClientConfiguration.DefaultConfiguration"/> for the client settings.
+    /// </summary>
+    /// <param name="url">URL of a resource where the V2 API is served.</param>
+    /// <param name="port">Port of the resource where the V2 API is served.</param>
+    public Client(string url, UInt16 port)
+        : this(url, port, ClientConfiguration.DefaultConfiguration) { }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="Client"/> class.
+    /// </summary>
+    /// <param name="url">URL of a resource where the V2 API is served.</param>
+    /// <param name="port">Port of the resource where the V2 API is served.</param>
+    /// <param name="timeout">The request timeout in seconds.</param>
+    /// <param name="secure">Flag indicating whether the client must use a secure connection.</param>
+    public Client(string url, UInt16 port, ulong timeout, bool secure)
+        : this(url, port, new ClientConfiguration(timeout, secure)) { }
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="Client"/> class.
+    /// </summary>
+    /// <param name="url">URL of a resource where the V2 API is served.</param>
+    /// <param name="port">Port of the resource where the V2 API is served.</param>
+    /// <param name="configuration">The configuration to use with this client.</param>
+    public Client(string url, UInt16 port, ClientConfiguration configuration)
+    {
+        GrpcChannelOptions options = new GrpcChannelOptions
+        {
+            Credentials = configuration.Secure
+                ? ChannelCredentials.SecureSsl
+                : ChannelCredentials.Insecure
+        };
+        _grpcChannel = GrpcChannel.ForAddress(url + ":" + port.ToString(), options);
+        _client = new Queries.QueriesClient(_grpcChannel);
+        _config = configuration;
+    }
```

Comment:
It could be a nice touch having a builder for this which took url and port as a minimum, and then the tls option (which I believe it is? maybe that also should be stated explicitly in the documentation) could be optionals to that builder. 

Also, isn't there something like a socket address instead of a raw string and the unsigned int?

Reply:
I added the socket address. Is a builder not something that can be added later if more parameters are added eventually? Currently it's just two. To my knowledge there's not a lombok equivalent in C#.